plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -119,6 +119,16 @@
     private final LoadingCache<String, Set<RoleGrant>> grantedPrincipalsCache;
     private final LoadingCache<String, Optional<String>> configValuesCache;
 
+    public static boolean isCacheDisabled(CachingHiveMetastoreConfig config)
+    {
+        return isCacheDisabled(config.getMetastoreCacheTtl(), config.getMetastoreCacheMaximumSize());
+    }
+
+    private static boolean isCacheDisabled(Duration cacheTtl, long maximumSize)
+    {
+        return cacheTtl.toMillis() == 0 || maximumSize == 0;
+    }
+
     public static HiveMetastore cachingHiveMetastore(HiveMetastore delegate, Executor executor, CachingHiveMetastoreConfig config)
```

Comment:
make it return `CachingHiveMatestore`

Reply:
Cant do it due to possibility of returning delegate from here as well https://github.com/trinodb/trino/blob/e127015b56bfe83f111f430e652a8f2530162ff9/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java#L136

Reply:
Yes. You need to fix callers to `cachingHiveMetastore`

Reply:
Ok so i will add separate commit then as this is asside of providing cache.

Reply:
Seems after changes this check is no longer needed in CachingHiveMetastore. So I've just removed it and made static factory methods to return CachingHiveMetastore